### PR TITLE
cmake: Switch from tri-state options to boolean. Stage FIVE. The GUI

### PR DESCRIPTION
**Pull request by hebasto**
Replaces multi-value `WITH_GUI` with boolean `BUILD_GUI` as discussed during the recent call.

The default value of the `BUILD_GUI` option is `OFF`.

Documentation has been adjusted.

While touching build docs, they were amended to use modern CMake invocations according to discussion in https://github.com/hebasto/bitcoin/pull/167 (should be trivial to review though).

### CODE REVIEW

**Comment by hebasto** [comment]
Rebased.

**Comment by hebasto** [comment]
Rebased.

**Comment by theuni** [comment]
Would you mind splitting this up into 2 commits: 1 for the logical change, and 1 for the rename to BUILD_GUI? That would make it easier to review imo.

**Comment by hebasto** [comment]
@theuni 
> Would you mind splitting this up into 2 commits: 1 for the logical change, and 1 for the rename to BUILD_GUI? That would make it easier to review imo.

Done.

**Comment by hebasto** [comment]
Rebased.

File: depends/toolchain.cmake.in [hunk 1]
```diff
@@ -117,11 +117,11 @@ unset(pkg_config_path)
 set(PKG_CONFIG_ARGN --static)
 
 
-# Ensure that the docstrings in the following `set(... CACHE ...)`
-# commands match those in the root CMakeLists.txt file.
-
-if(NOT WITH_GUI AND "@no_qt@" STREQUAL "1")
-  set(WITH_GUI OFF CACHE STRING "Build GUI.")
+# Set configuration options for the main build system.
+if("@no_qt@")
```
Comment (theuni) [inline depends/toolchain.cmake.in:121]: I wonder if we should depart from upstream/master behavior for all of these options to be more CMake-like?

(I'm not just referring to qt here, but all depends options)

What if depends doesn't turn anything on by default? Meaning.. quit forwarding any meaning from built packages into Core's build. So, you can build depends with any set of packages/options, but that doesn't turn anything just on by using the generated toolchain file.

That way, if I build depends with qt, I still have to specify `WITH_GUI` in order to get bitcoin-qt. Conversely, if I build depends with `NO_QT` and build with `WITH_GUI`, that would be an error.

Thoughts? @hebasto @fanquake @laanwj

Comment (hebasto) [reply depends/toolchain.cmake.in:121]: > ... to be more CMake-like?

In the future, CMake can manage all dependencies using its own modules like `ExternalProject` or `FetchContent`.

**Comment by hebasto** [comment]
Rebased, which made the last commit simpler.

**Review by TheCharlatan — APPROVED** [review]
lgtm

**Pull request merged**